src/libexpr/primops/fetchTree.cc

```diff
@@ -193,8 +193,52 @@ static void prim_fetchTree(EvalState & state, const PosIdx pos, Value * * args,
     fetchTree(state, pos, args, v, std::nullopt, FetchTreeParams { .allowNameArgument = false });
 }
 
-// FIXME: document
-static RegisterPrimOp primop_fetchTree("fetchTree", 1, prim_fetchTree);
+static RegisterPrimOp primop_fetchTree({
+    .name = "fetchTree",
+    .args = {"args"},
+    .doc = R"(
+      Fetch a source tree or a plain file using one of supported
+      backends. *args* can be an attribute set representation of
+      [flake reference](../command-ref/new-cli/nix3-flake.html#flake-references)
+      in which case the referenced tree is fetched. Otherwise, it can
+      be an URL.
+
+      Here are some examples of how to use `fetchTree`:
+
+      - To fetch a GitHub repository use:
+
+          ```nix
+          builtins.fetchTree {
+            type = "github";
+            owner = "NixOS";
+            repo = "nixpkgs";
+          }
+          ```
+
+        This evaluates to an attribute set similar to the following:
+
+          ```
+          {
+            lastModified = 1686503798; lastModifiedDate = "20230611171638";
+            narHash = "sha256-rA9RqKP9OlBrgGCPvfd5HVAXDOy8k2SmPtB/ijShNXc=";
+            outPath = "/nix/store/l5m6qlvfs9sdw14ja3qbzpglcjlb6j1x-source";
+            rev = "ae2e6b3958682513d28f7d633734571fb18285dd"; shortRev = "ae2e6b3";
+          }
+          ```
+      - To fetch a single file from an URL use:
+
+          ```nix
+          builtins.fetchTree "https://example.com/"
+          ```
+
+      This function requires the experimental feature `flakes` to be
+      enabled. In [pure evaluation
+      mode](../command-ref/conf-file.html#conf-pure-eval) (which is
+      the default) it requires the flake reference to be locked in
+      `flake.lock`.
```

Comment:
I don't think this is correct. It just requires an immutable flake ref, but it doesn't need to be in `flake.lock`.